Index: scst-3.4.x/scst/src/dev_handlers/scst_vdisk.c
===================================================================
--- scst-3.4.x.orig/scst/src/dev_handlers/scst_vdisk.c
+++ scst-3.4.x/scst/src/dev_handlers/scst_vdisk.c
@@ -73,8 +73,8 @@
 #include "scst_dev_handler.h"
 
 /* 8 byte ASCII Vendor */
-#define SCST_FIO_VENDOR			"SCST_FIO"
-#define SCST_BIO_VENDOR			"SCST_BIO"
+#define SCST_FIO_VENDOR			"BAUM"
+#define SCST_BIO_VENDOR			"BAUM"
 /* 4 byte ASCII Product Revision Level - left aligned */
 #define SCST_FIO_REV			"340 "
 
