Index: scst-3.4.x/qla2x00t-32gbit/qla2x00-target/scst_qla2xxx.c
===================================================================
--- scst-3.4.x.orig/qla2x00t-32gbit/qla2x00-target/scst_qla2xxx.c
+++ scst-3.4.x/qla2x00t-32gbit/qla2x00-target/scst_qla2xxx.c
@@ -1516,6 +1516,7 @@ static int sqa_xmit_response(struct scst
 	cmd->offset = scst_cmd_get_ppl_offset(scst_cmd);
 	cmd->scsi_status = scst_cmd_get_status(scst_cmd);
 	cmd->cdb = (unsigned char *) scst_cmd_get_cdb(scst_cmd);
+	cmd->cdb_len = scst_cmd_get_cdb_len(scst_cmd);
 	cmd->lba = scst_cmd_get_lba(scst_cmd);
 	cmd->trc_flags |= TRC_XMIT_STATUS;
 
@@ -1599,7 +1600,8 @@ static int sqa_rdy_to_xfer(struct scst_c
 	cmd->dma_data_direction =
 		scst_to_tgt_dma_dir(scst_cmd_get_data_direction(scst_cmd));
 
-	cmd->cdb = scst_cmd_get_cdb(scst_cmd);
+	cmd->cdb = (unsigned char *) scst_cmd_get_cdb(scst_cmd);
+	cmd->cdb_len = scst_cmd_get_cdb_len(scst_cmd);
 	cmd->sg = scst_cmd_get_sg(scst_cmd);
 	cmd->sg_cnt = scst_cmd_get_sg_cnt(scst_cmd);
 	cmd->scsi_status = scst_cmd_get_status(scst_cmd);
Index: scst-3.4.x/qla2x00t-32gbit/qla_target.c
===================================================================
--- scst-3.4.x.orig/qla2x00t-32gbit/qla_target.c
+++ scst-3.4.x/qla2x00t-32gbit/qla_target.c
@@ -2404,7 +2404,19 @@ static int qlt_pci_map_calc_cnt(struct q
 {
 	struct qla_tgt_cmd *cmd = prm->cmd;
 
-	BUG_ON(cmd->sg_cnt == 0);
+	if (0 == cmd->sg_cnt) {
+		int i;
+
+		printk("bufflen = %u, scsi_status = %d, cdb_len = %d\nCDB buf: ", cmd->bufflen, cmd->scsi_status, cmd->cdb_len);
+
+		for (i = 0; i < cmd->cdb_len; ++i)
+			printk("%02x ", cmd->cdb[i]);
+		printk("\n");
+
+		BUG_ON(cmd->sg_cnt == 0);
+		//WARN_ON_ONCE(cmd->sg_cnt == 0);
+		//goto out_err;
+	}
 
 	prm->sg = (struct scatterlist *)cmd->sg;
 	prm->seg_cnt = dma_map_sg(&cmd->qpair->pdev->dev, cmd->sg,
Index: scst-3.4.x/qla2x00t-32gbit/qla_target.h
===================================================================
--- scst-3.4.x.orig/qla2x00t-32gbit/qla_target.h
+++ scst-3.4.x/qla2x00t-32gbit/qla_target.h
@@ -960,6 +960,7 @@ struct qla_tgt_cmd {
 
 	struct crc_context *ctx;
 	const uint8_t	*cdb;
+	unsigned int cdb_len;
 	uint64_t	lba;
 	uint16_t	a_guard, e_guard, a_app_tag, e_app_tag;
 	uint32_t	a_ref_tag, e_ref_tag;
